integration/test/metric_value_benchmark/cpu_test.go

```diff
@@ -0,0 +1,77 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build linux && integration
+// +build linux,integration
+
+package metric_value_benchmark
+
+import (
+	"time"
+
+	"github.com/aws/amazon-cloudwatch-agent/integration/test/metric"
+	"github.com/aws/amazon-cloudwatch-agent/integration/test/status"
+)
+
+const cpuTestName = "CPU"
+
+type CPUTestRunner struct {
+}
+
+var metricsToFetch = []string{
+	"cpu_time_active", "cpu_time_guest", "cpu_time_guest_nice", "cpu_time_idle", "cpu_time_iowait", "cpu_time_irq",
+	"cpu_time_nice", "cpu_time_softirq", "cpu_time_steal", "cpu_time_system", "cpu_time_user",
+	"cpu_usage_active", "cpu_usage_quest", "cpu_usage_quest_nice", "cpu_usage_idle", "cpu_usage_iowait",
+	"cpu_usage_irq", "cpu_usage_nice", "cpu_usage_softirq", "cpu_usage_steal", "cpu_usage_system", "cpu_usage_user"}
+
+func (t *CPUTestRunner) validate() status.TestGroupResult {
+	testResults := []status.TestResult{}
+	for _, metricName := range metricsToFetch {
+		testResult := validateCpuMetric(metricName)
+		testResults = append(testResults, testResult)
+	}
+
+	return status.TestGroupResult{
+		Name:        t.getTestName(),
+		TestResults: testResults,
+	}
+}
+
+func (t *CPUTestRunner) getTestName() string {
+	return cpuTestName
+}
+
+func (t *CPUTestRunner) getAgentConfigFileName() string {
+	return agentConfigFileName
+}
+
+func (t *CPUTestRunner) getAgentRunDuration() time.Duration {
+	return minimumAgentRuntime
+}
+
+func validateCpuMetric(metricName string) status.TestResult {
+	testResult := status.TestResult{
+		Name:   metricName,
+		Status: status.FAILED,
+	}
+
+	fetcher, err := metric.GetMetricFetcher(metricName)
+	if err != nil {
+		return testResult
+	}
+
+	values, err := fetcher.Fetch(namespace, metricName, metric.AVERAGE)
```

Comment:
Assuming this is actually making a call to CW, would making one call to fetch all metrics be better as opposed to fetching one at a time?

Reply:
Sacrificing api cost for logic/code cleanness. If we want to we can just use the provided interface (which takes a lot of query requests at once) but it makes code harder to understand/harder to think of each test independently. Given that api cost isn't a big deal for this test package, I think we can change this if this becomes a problem

Reply:
Would the logic change a lot? Id imagine only the initial call to backend would be done once but the rest of the stuff like validating each metric can remain the same... just that it would leverage what was already retrieved from the backend at that point.

Reply:
well it wouldn't read the same. right now logic's clean with validate() call simply calling all validateCPUMetric with each metric. It's a higher level of abstraction. What is exactly the concern? Performance? How concerned are we about test run's performance?

Reply:
Yeah probably only a small bit from performance standpoint and test runtime since its more network calls. But this isnt a blocker if others are fine with it.